File: PHP/7php 1/incrementer.php
<?php
/*
|--------------------------------------------------------------------------
| Incrementer
|--------------------------------------------------------------------------
|
*/

function incrementer($nums) {
   $res = [];
   for ($i=0; $i < count($nums); $i++) { 
      array_push($res, ($nums[$i] + $i + 1) % 10);
   }
   return $res;
}


function incrementer($nums) {
   foreach ($nums as $i => &$n) {
      $n = ($n + $i + 1) % 10;
   }
   return $nums;
}


function incrementer($nums) {
   return array_map(function ($e, $i) {
      return ($e+$i+1)%10;
   }, $nums, array_keys($nums));
}


print_r(incrementer([1, 2, 3]));
print_r(incrementer([4, 6, 7, 1, 3]));

File: Python/7py 2/incrementer.py
"""
|--------------------------------------------------------------------------
| Incrementer
|--------------------------------------------------------------------------
|
"""

def incrementer(nums):
   res = []
   for i, v in enumerate(nums):
      res.append((v+i+1) % 10)
   return res


def incrementer(nums):
   return [(v+i)%10 for i,v in enumerate(nums,1)]


def incrementer(nums):
   return [int(str((a+i+1))[-1]) for a, i in enumerate(nums)]


print(incrementer([]), [])
print(incrementer([1, 2, 3]), [2, 4, 6])
print(incrementer([4, 6, 7, 1, 3]), [5, 8, 0, 5, 8])
print(incrementer([3, 6, 9, 8, 9]), [4, 8, 2, 2, 4])